rsocket-core/src/main/java/io/rsocket/core/RSocketRequester.java

```diff
@@ -120,12 +121,15 @@
     this.allocator = connection.alloc();
     this.payloadDecoder = payloadDecoder;
     this.errorConsumer = errorConsumer;
-    this.streamIdSupplier = streamIdSupplier;
     this.mtu = mtu;
     this.leaseHandler = leaseHandler;
     this.senders = new SynchronizedIntObjectHashMap<>();
     this.receivers = new SynchronizedIntObjectHashMap<>();
     this.onClose = MonoProcessor.create();
+    this.streamIdSupplier =
+        isResumeEnabled
+            ? new ResumableStreamIdSupplier(streamIdSupplier, receivers)
+            : streamIdSupplier;
```

Comment:
I guess this should not be the case since if we going to come to http2 transport, then once we get it overflowed - everything will be broken. 

Spec also mentions that reuse of stream-id is recommended only with resumability -> https://github.com/rsocket/rsocket/blob/master/Protocol.md#lifetime.

@linux-china, @jjeffcaii do you know folks, whether it should be re-usable id all the time, I remember you were asking about that, but I'm not sure that it is always needed.

@rstoyanchev Maybe we can do a configuration?

Reply:
Alright, I guess in this PR I will just provide a relaxed stream id generation. No need to care about all the things right now

Reply:
@OlegDokuka Yes, SteamID is re-usable by default in current Go and Rust SDK.